compiler/ballerina-treegen/src/main/resources/syntax_tree_descriptor.json

```diff
@@ -1125,6 +1125,56 @@
                 }
             ]
         },
+        {
+            "name": "ObjectConstructorExpressionNode",
+            "base": "ExpressionNode",
+            "kind": "OBJECT_CONSTRUCTOR",
+            "attributes": [
+                {
+                    "name": "annotations",
+                    "type": "AnnotationNode",
+                    "occurrences": "MULTIPLE"
+                },
+                {
+                    "name": "objectTypeQualifier",
+                    "type": "Token",
+                    "isOptional": true
```

Comment:
```json
                {
                    "name": "qualifierList",
                    "type": "Token",
                    "occurrences": "MULTIPLE"
                }
```
Shall we use qualifierList instead?  bcz when there's a change to no of preceding qualifiers we don't need to change the syntax tree. 
Check `MethodDeclarationNode` there's similar case with `remote-qual := remote`

Reply:
+1

Reply:
I changed this. Accept as a qualifier list (not optional) and throw an exception on Node transformer for unexpected tokens hope this the correct way to handle this.